server/jetstream_api.go

```diff
@@ -1234,6 +1234,41 @@ func (s *Server) jsStreamCreateRequest(sub *subscription, c *client, _ *Account,
 		return
 	}
 
+	// If we have a republish directive check if we can create a transform here.
```

Comment:
Agree with @kozlovic , but it's this code that should move

I'd move this entire block into `func (s *Server) checkStreamCfg(config *StreamConfig, acc *Account) (StreamConfig, *ApiError)` `stream.go` line `858` or below.
This is the place for stream config checks like that.
In the current form restore would not check this.
Probably shortens the change as the error response part is dealt with by the caller already.

The previous check for `sealed` above only exists as it explicitly applies to create only.

Reply:
Yep done..